pkg/controller/nodefeaturediscovery/nodefeaturediscovery_controls.go

```diff
@@ -147,18 +131,13 @@ func ClusterRoleBinding(n NFD) (ResourceStatus, error) {
 	state := n.idx
 	obj := n.resources[state].ClusterRoleBinding
 
-	if len(n.ins.Spec.OperandNamespace) != 0 {
-		obj.SetNamespace(n.ins.Spec.OperandNamespace)
-	}
-
 	found := &rbacv1.ClusterRoleBinding{}
 	logger := log.WithValues("ClusterRoleBinding", obj.Name, "Namespace", obj.Namespace)
 
-	if err := controllerutil.SetControllerReference(n.ins, &obj, n.rec.scheme); err != nil {
```

Comment:
How did you decide which objects needed references?

Reply:
Namespaced resource cannot be a controllerreference for cluster-wide resources.